SyslogNet.Client/Transport/SyslogLocalSender.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using SyslogNet.Client.Serialization;
+
+namespace SyslogNet.Client.Transport
+{
+	public class SyslogLocalSender : ISyslogMessageSender, IDisposable
+	{
+		// Will not work on Windows, as this relies on Unix system calls
+		public SyslogLocalSender()
+		{
+			PlatformID platform = Environment.OSVersion.Platform;
+			if (!(platform == PlatformID.MacOSX || platform == PlatformID.Unix)) {
+				throw new CommunicationsException("SyslogLocalSender is only available on Unix-like systems (e.g., Linux, BSD, OS X)");
+			}
+		}
+
+		[DllImport("libc", ExactSpelling=true)]

```

Comment:
Sure. That extra layer of indirection would probably provide insurance in case of theoretical future platforms that don't have a libc library but do, somehow, implement the `openlog()` / `syslog()` / `closelog()` API. If Mono is ported to those platforms, it will probably implement those calls the right way. I did it this way because [that's how log4net does it](https://apache.googlesource.com/log4net/+/trunk/src/log4net/Appender/LocalSyslogAppender.cs), but that's certainly not the only way to do it.

I'll rewrite the code to use Mono's Mono.Unix.Native.Syscall instead.